### PR TITLE
Flutter SDK

### PR DESCRIPTION
**Pull request by laves**

### CODE REVIEW

File: demo/flutter/lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+    String contextPath = await _extractAsset(contextAsset);
+
+    try {
+      _picovoiceManager = await PicovoiceManager.create(
+          keywordPath, wakeWordCallback, contextPath, inferenceCallback,
+          errorCallback: errorCallback);
+    } on PvError catch (ex) {
+      print("Failed to initialize Picovoice: ${ex.message}");
+    } finally {
+      this.setState(() {
```
Comment (kenarsa) [inline demo/flutter/lib/main.dart:66]: don't enable unless success

File: demo/flutter/lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    String contextAsset =
+        "assets/contexts/$platform/smart_lighting_$platform.rhn";
+    String contextPath = await _extractAsset(contextAsset);
+
+    try {
+      _picovoiceManager = await PicovoiceManager.create(
+          keywordPath, wakeWordCallback, contextPath, inferenceCallback,
+          errorCallback: errorCallback);
+    } on PvError catch (ex) {
+      print("Failed to initialize Picovoice: ${ex.message}");
```
Comment (kenarsa) [inline demo/flutter/lib/main.dart:64]: let's see if flutter has an error reporting mechanism

File: demo/flutter/lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+      });
+    }
+  }
+
+  void inferenceCallback(Map<String, dynamic> inference) {
+    this.setState(() {
+      rhinoText = prettyPrintInference(inference);
+    });
+
+    Future.delayed(const Duration(milliseconds: 2500), () {
```
Comment (kenarsa) [inline demo/flutter/lib/main.dart:85]: handle the case if they say something while waiting

File: demo/flutter/lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+      await _picovoiceManager.start();
+      this.setState(() {
+        isProcessing = true;
+        rhinoText = "Listening for wake word...";
+      });
+    } on PvAudioException catch (ex) {
+      print("Failed to start audio capture: ${ex.message}");
+    } finally {
+      this.setState(() {
+        isButtonDisabled = false;
```
Comment (kenarsa) [inline demo/flutter/lib/main.dart:154]: only enable after success

File: demo/flutter/lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    });
+
+    try {
+      await _picovoiceManager.stop();
+      this.setState(() {
+        isProcessing = false;
+        rhinoText = "";
+      });
+    } on PvAudioException catch (ex) {
+      print("Failed to start audio capture: ${ex.message}");
```
Comment (kenarsa) [inline demo/flutter/lib/main.dart:175]: ailed to start audio -> ailed to stop[ audio 

can this even fail?

Comment (laves) [reply demo/flutter/lib/main.dart:175]: yep!

**Pull request merged**